### ISSUE TITLE
AI can use one creature to attack multiple targets (player and planeswalkers)

### ISSUE THREAD

**Issue by copperfield42** [magefree/mage#7434]
![image](https://user-images.githubusercontent.com/4504719/105616901-9f1c6e80-5db0-11eb-9467-73fc404f7f21.png)

the AI was able to declare attack with the same creature as it were multiple creatures an attacked all my Chandras and when I blocked only one of those multiples attack was blocked

```

07:20 PM: Turn 41 Computer 3 (34 - -2 - 40 - 7)
07:20 PM, T41.UN: Computer 3 untapped Kalemne, Disciple of Iroas [dca]
07:20 PM, T41.DR: Computer 3 draws a card
07:20 PM, T41.M1: Computer 3 activates: Equip. from Loxodon Warhammer [169] targeting Kalemne, Disciple of Iroas [dca]
07:20 PM, T41.M1: Computer 3 plays Mountain [751]
07:20 PM, T41.M1: Computer 3 puts Mountain [751] from hand onto the Battlefield
07:20 PM, T41.DA: Computer 3 attacks with 3 creatures
07:25 PM, T41.DB: Attacked player: copperfield
07:25 PM, T41.DB: Attacker: Kalemne, Disciple of Iroas [dca] (6/3) blocked by Dragon Egg [056] (0/2) Dragon Egg [1be] (0/2) Dragon Egg [3df] (0/2) Dragon Egg [87e] (0/2) Dragon Egg [f64] (0/2) Goblin [c50] (1/1) Goblin [307] (1/1) Goblin [dfb] (1/1) 
07:25 PM, T41.DB: Attacker: Kalemne, Disciple of Iroas [dca] (6/3) unblocked
07:25 PM, T41.DB: Attacker: Kalemne, Disciple of Iroas [dca] (6/3) unblocked
07:25 PM, T41.FCD: Kalemne, Disciple of Iroas [dca] deals 6 damage to Chandra, Bold Pyromancer [9da]
07:25 PM, T41.FCD: Kalemne, Disciple of Iroas [dca] deals 6 damage to Chandra, Flame's Fury [54e]
07:25 PM, T41.FCD: Computer 3 gains 18 life
07:25 PM, T41.FCD: Kalemne, Disciple of Iroas [dca] deals 2 damage to Dragon Egg [056]
07:25 PM, T41.FCD: Kalemne, Disciple of Iroas [dca] deals 2 damage to Dragon Egg [1be]
07:25 PM, T41.FCD: Kalemne, Disciple of Iroas [dca] deals 2 damage to Dragon Egg [3df]
07:25 PM, T41.FCD: Chandra, Bold Pyromancer [9da] is put into graveyard from battlefield
07:25 PM, T41.FCD: Chandra, Flame's Fury [54e] is put into graveyard from battlefield
07:25 PM, T41.FCD: Dragon Egg [1be] died
07:25 PM, T41.FCD: Dragon Egg [3df] died
07:25 PM, T41.FCD: Dragon Egg [056] died
07:25 PM, T41.CD: Kalemne, Disciple of Iroas [dca] deals 1 damage to Goblin [307]
07:25 PM, T41.CD: Kalemne, Disciple of Iroas [dca] deals 2 damage to Dragon Egg [87e]
07:25 PM, T41.CD: Kalemne, Disciple of Iroas [dca] deals 1 damage to Goblin [c50]
07:25 PM, T41.CD: Kalemne, Disciple of Iroas [dca] deals 2 damage to Dragon Egg [f64]
07:25 PM, T41.CD: Computer 3 gains 6 life
07:25 PM, T41.CD: Dragon Egg [87e] died
07:25 PM, T41.CD: Goblin [307] died
07:25 PM, T41.CD: Kalemne, Disciple of Iroas [dca] died
07:25 PM, T41.CD: Dragon Egg [f64] died
07:25 PM, T41.CD: Goblin [c50] died
07:25 PM, T41.CD: Computer 3 puts Kalemne, Disciple of Iroas [dca] from exile zone to his or her command zone
07:25 PM, T41.CD: Goblin [dfb] deals 3 damage to Kalemne, Disciple of Iroas [dca]
07:25 PM, T41.CD: Goblin [307] deals 3 damage to Kalemne, Disciple of Iroas [dca]
07:25 PM, T41.CD: Goblin [c50] deals 3 damage to Kalemne, Disciple of Iroas [dca]
```

**Comment by copperfield42** [comment]
![005 multi attack 4](https://user-images.githubusercontent.com/4504719/107127456-946fd800-688c-11eb-99e0-6e4cdf4102c5.PNG)

It happens again, this time attacking another player and my planeswalker, and I couldn't block it

**Comment by JayDi85** [comment]
Is that a rare bug or it appears in all AI games?

**Comment by JayDi85** [comment]
It can be related to the changes from a7e1c501224d16939170a6537ade84c0c490c8a1 (and later commits) about AI and planeswalker attacks. I don't check current code but first commit version allows attack multiple planeswalkers by one creature.

Maybe @jeffwadsworth can review that code for more details.

**Comment by copperfield42** [comment]
> Is that a rare bug or it appears in all AI games?

is rare, the screenshot I show are the only times I watched this bug, I had planeswalker out in other occasions and that didn't happens

**Comment by copperfield42** [comment]
Here is another sample, this time 2 creatures are doing the multi-attack
![005 multi attack 8](https://user-images.githubusercontent.com/4504719/142695304-3110d291-a105-4f5d-a6b3-38dfa7d590a7.PNG)

is a little interesting that it trigger Circle of Flames multiple times

**Comment by copperfield42** [comment]
another sample
![image](https://user-images.githubusercontent.com/4504719/153522511-f5dc390b-9816-48f7-a53b-b40b605b76b7.png)


there seem to be a correlation with the ability of vigilance and this bug

**Comment by jeffwadsworth** [comment]
I will test the Vigilance thing because I have never seen this bug during playtesting.  Hopefully, that is the weird link.

**Comment by Alex-Vasile** [comment]
Another reported occurance:
![unknown-1](https://user-images.githubusercontent.com/48962821/180474560-4f62db6f-1f6f-4809-95f5-821d62728d99.png)

**Comment by JayDi85** [comment]
Multiple targets attack by AI fixed in e4157fefb8d924cbf014534753e13c288d2d902f. Real reason was in miss attacking status, see previous comments https://github.com/magefree/mage/issues/7434#issuecomment-774707516.
